src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -163,8 +166,13 @@ public int numberOfShards() {
     }
 
     @Override
-    public UnmodifiableIterator<IndexShard> iterator() {
-        return shards.values().iterator();
+    public Iterator<IndexShard> iterator() {
```

Comment:
Since `shards` is an ImmutableMap, can this return an `UnmodifiableIterator` again to indicate that no one should be changing things?